### PR TITLE
fix: make sure all cookies expire at the same point in time

### PR DESCRIPTION
**Pull request by m90**
Ticket https://phabricator.wikimedia.org/T350469

Using this patch, all cookies will expire at the same time (i.e. 3 months TTL).

### CODE REVIEW

File: public_html/index.php [hunk 1]
```diff
@@ -33,7 +33,9 @@
 */
 
 require_once ( 'php/Widar.php' ) ;
-$widar = new Widar ( 'widar' ) ;
+$widar = new Widar ( 'widar', function () {
+    return time()+60*60*24*30*3; // make all cookies expire in three months
```
[the remark this answers is not in the document]
Comment (tarrow) [reply public_html/index.php:37]: This would also look fine to me but I wouldn't block merging for this.

File: public_html/index.php [hunk 2]
```diff
@@ -33,7 +33,9 @@
 */
 
 require_once ( 'php/Widar.php' ) ;
-$widar = new Widar ( 'widar' ) ;
+$widar = new Widar ( 'widar', function () {
```
Comment (tarrow) [inline public_html/index.php:36]: I guess this now needs to be tweaked to take the plain value rather than a function

**Comment by tarrow** [comment]
This should be merged after the patch in wbstack/magnustools

File: public_html/index.php (hunk 2 above)
Comment (m90) [reply public_html/index.php]: Done so.

**Comment by m90** [comment]
Running the latest version locally, I get 3 cookies with the same lifetime now

image

**Review by tarrow — APPROVED** [review]
Looks good, I shied away from testing this fully locally since the build process was clearly going to take me a while.

**Pull request merged**